lumibot/tools/polygon_helper.py

```diff
@@ -160,6 +165,45 @@ def get_price_data_from_polygon(
     return df_all
 
 
+def validate_cache(force_cache_update: bool, asset: Asset, cache_file: Path, api_key: str):
+    """
+    If the list of splits for a stock have changed then we need to invalidate its cache
+    because all of the prices will have changed (because we're using split adjusted prices).
+    Get the splits data from Polygon only once per day per stock.
+    Use the timestamp on the splits feather file to determine if we need to get the splits again.
+    When invalidating we delete the cache file and return force_cache_update=True too.
+    """
+    if asset.asset_type != Asset.AssetType.STOCK:
+        return force_cache_update
+    cached_splits = pd.DataFrame()
+    splits_file_stale = True
+    splits_file_path = Path(str(cache_file).rpartition(".feather")[0] + "_splits.feather")
+    if splits_file_path.exists():
+        splits_file_stale = datetime.fromtimestamp(splits_file_path.stat().st_mtime).date() != date.today()
+        if splits_file_stale:
+            cached_splits = pd.read_feather(splits_file_path)
+    if splits_file_stale or force_cache_update:
+        polygon_client = RESTClient(api_key)
+        # Need to get the splits in execution order to make the list comparable across invocations.
+        splits = polygon_client.list_splits(ticker=asset.symbol, sort="execution_date", order="asc")
+        if isinstance(splits, Iterator):
+            # Convert the generator to a list so DataFrame will make a row per item.
+            splits_df = pd.DataFrame(list(splits))
```

Comment:
This code does not assume the splits will be a dataframe, it assumes it will be an Iterator[Split] as specified by the RESTClient.list_splits method signature.